src/D2L.CodeStyle.Analyzers/Async/Generator/SyntaxTransformer.cs

```diff
@@ -0,0 +1,120 @@
+﻿#nullable enable
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace D2L.CodeStyle.Analyzers.Async.Generator;
+
+/// <summary>
+/// Base class for things that transform a SyntaxNode into another and collect
+/// diagnostics. This is useful for writing generators that copy+paste code
+/// with only light modifications.
+/// </summary>
+public abstract class SyntaxTransformer {
+	protected readonly SemanticModel m_model;
+	private readonly ImmutableArray<Diagnostic>.Builder m_diagnostics = ImmutableArray.CreateBuilder<Diagnostic>();
+
+	/// <summary>
+	/// A name for the generator that will be shown to the user if we crash
+	/// </summary>
+	protected abstract string Name { get; }

```

Comment:
Not used in this class FYI

Reply:
Used in `GeneratorError`

https://github.com/Brightspace/D2L.CodeStyle/pull/785/files#diff-c358c76ce31b7970f2ed441e3a685108b80e12e0f0cb42e468139b51e1cc550dR77

Reply:
My bad

Reply:
I might change this anyway